src/lib.rs

```diff
@@ -1 +1,192 @@
 #![allow(non_snake_case)]
+
+use std::ffi::{c_void, OsStr, OsString};
+use std::mem::size_of;
+use std::path::{Path, PathBuf};
+use thiserror::Error;
+use windows::Win32::Foundation::{CloseHandle, GetLastError, PWSTR};
+use windows::Win32::Security::SECURITY_ATTRIBUTES;
+use windows::Win32::System::Threading::{
+    GetExitCodeProcess, TerminateProcess, WaitForSingleObject, PROCESS_CREATION_FLAGS,
+    PROCESS_INFORMATION, STARTUPINFOW,
+};
+use windows::Win32::System::WindowsProgramming::INFINITE;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("cannot create process: {0}")]
+    CreationFailed(u32),
+    #[error("cannot wait process: {0}")]
+    WaitFailed(u32),
+    #[error("cannot kill process: {0}")]
+    KillFailed(u32),
+}
+
+type Result<T> = std::result::Result<T, Error>;
+
+#[derive(Debug)]
+pub struct Command {
+    command: OsString,
+    inherit_handles: Option<bool>,
+    current_directory: Option<PathBuf>,
+}
+
+impl Command {
+    pub fn new(command: impl AsRef<OsStr>) -> Command {
+        Command {
+            command: command.as_ref().to_owned(),
+            inherit_handles: None,
+            current_directory: None,
+        }
+    }
+
+    pub fn inherit(mut self, bool: bool) -> Command {
+        self.inherit_handles = Some(bool);
+        self
+    }
+
+    pub fn current_dir(mut self, dir: impl AsRef<Path>) -> Command {
```

Comment:
1. You need to return `&mut Self` instead of a new instance to match the std::process API (everywhere).
2. Again, since your end goal is to own the value, you need to use `impl Into<PathBuf>` and not `AsRef`